packages/datadog-shimmer/src/shimmer.js

```diff
@@ -3,20 +3,24 @@
 const log = require('../../dd-trace/src/log')
 
 function copyProperties (original, wrapped) {
-  // TODO getPrototypeOf is not fast. Should we instead do this in specific
-  // instrumentations where needed?
-  const proto = Object.getPrototypeOf(original)
-  if (proto !== Function.prototype) {
-    Object.setPrototypeOf(wrapped, proto)
+  if (original.constructor !== Function) {
+    const proto = Object.getPrototypeOf(original)
+    if (proto !== Function.prototype) {
+      Object.setPrototypeOf(wrapped, proto)
+    }
   }
-  const props = Object.getOwnPropertyDescriptors(original)
-  const keys = Reflect.ownKeys(props)
```

Comment:
since you don't iterate on `ownKeys` anymore, is there absolutely no difference between the list given by `ownKeys()`, and the one given by `getOwnPropDes...()` ?
Just giving a made up example but maybe `ownKeys()` doesn't return non-enumerable props, but `getOwnPropDesc()` does ?

Reply:
Even if that's the case, it might not be a problem, to also copy non-enumerable props for example, but it might still  be something to double check

Reply:
`Reflect.ownKeys()` returns all keys, no matter if they are enumerable or not. That is however not important in this case due to only getting the keys of the object returned by `getOwnPropertyDescriptors()` and that object only contains enumerable properties.

Your comment made me read the code again and it did indeed contain an issue! So thank you for that! I accidentally used Object.entries() in the fallback and that is wrong (it would skip the symbol properties). I'll also add a test case to address that.

For questions around native APIs, I generally recommend to check [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/ownKeys) (it is easier to read than the spec) :-)

Reply:
Ah I'm glad because I also was looking at the `Object.entries()` with suspicious eyes!